src/fundamental/sbat.h

```diff
@@ -2,8 +2,12 @@
 
 #ifdef SBAT_DISTRO
 #  include "version.h"
-#  define SBAT_SECTION_TEXT \
+#  define SBAT_BOOT_SECTION_TEXT \
         "sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md\n" \
-        SBAT_PROJECT ",1,The systemd Developers," SBAT_PROJECT "," PROJECT_VERSION "," PROJECT_URL "\n" \
-        SBAT_PROJECT "." SBAT_DISTRO "," STRINGIFY(SBAT_DISTRO_GENERATION) "," SBAT_DISTRO_SUMMARY "," SBAT_DISTRO_PKGNAME "," SBAT_DISTRO_VERSION "," SBAT_DISTRO_URL "\n"
+        SBAT_PROJECT "-boot" ",1,The systemd Developers," SBAT_PROJECT "," PROJECT_VERSION "," PROJECT_URL "\n" \
+        SBAT_PROJECT "-boot" "." SBAT_DISTRO "," STRINGIFY(SBAT_DISTRO_GENERATION) "," SBAT_DISTRO_SUMMARY "," SBAT_DISTRO_PKGNAME "," SBAT_DISTRO_VERSION "," SBAT_DISTRO_URL "\n"
+#  define SBAT_STUB_SECTION_TEXT \
+        "sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md\n" \
+        SBAT_PROJECT "-stub" ",1,The systemd Developers," SBAT_PROJECT "," PROJECT_VERSION "," PROJECT_URL "\n" \
+        SBAT_PROJECT "-stub" "." SBAT_DISTRO "," STRINGIFY(SBAT_DISTRO_GENERATION) "," SBAT_DISTRO_SUMMARY "," SBAT_DISTRO_PKGNAME "," SBAT_DISTRO_VERSION "," SBAT_DISTRO_URL "\n"
```

Comment:
hmm, shouldn't the generic project names also stick around, already for compat reasons? i mean, sbat info is already included in a released version of systemd, so we better progress with the existing names?

my understanding is that it should rather be too verbose than too finegrained, so that you can precisely revoke what you want?

Reply:
The 4th field is used to list the generic project name. I don't think we need to keep the same component name, as the revocation payload can mention any, so a one-time switch to a more granular definition should be ok.

Reply:
well, still not convinced we should change things so willnilly here. People might use sbat to simply check if something is used in a UKI, and this will now break if we keep renaming our sbat parts...

Reply:
That should have its own identifier, and that's what we tell people to do. After all we clearly document that sd-stub is only one possible implementation of a stub, and it's perfectly fine to use a different one, that thus won't mention 'systemd' at all - but it would still be a UKI.

Reply:
i am not sure I follow. if people have a PE and want to check if this contains systemd code, things now fall apart, they have to update their rules.

To me that smells like compat breakage. SBAT seems useless if we just keep dropping lines from it randomly, in particular if no history is kept anywhere about the precise identifiers we use or used.

Maybe I am not understanding what sbat is supposed to be (which I wouldnt surprised by, the spec is not a great read), but my understanding was that you have recognizable, stable identifiers of components, to which you then can assign versions numbers. If that's not the case then I am puzzled by all of this.

Reply:
It's not really a compat break - it simply means that if we have a vulnerability found after this is merged, but that is pre-existing, then the revocation will have to revoke both old and new, ie 2 lines instead of one in the csv. It might happen, but on balance, I'd rather chance that, and be able to have separate revocations for sd-stub and sd-boot, otherwise a revocation for sd-boot will automatically result in all ukis using sd-stub to be revoked too.